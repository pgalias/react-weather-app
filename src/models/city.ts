export interface City {
  city: string;
  country: string;
  state: string;
}
