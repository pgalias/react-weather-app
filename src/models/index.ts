export * from './location';
export * from './units';
