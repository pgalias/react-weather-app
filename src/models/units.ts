export enum Units {
  airPressureAtSeaLevel = 'air_pressure_at_sea_level',
  airTemperature = 'air_temperature',
  cloudAreaFraction = 'cloud_area_fraction',
  relativeHumidity = 'relative_humidity',
  windFromDirection = 'wind_from_direction',
  windSpeed = 'wind_speed',
}

export type UnitsNames = keyof typeof Units;
