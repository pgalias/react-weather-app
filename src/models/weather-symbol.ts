export enum WeatherSymbol {
  'clearsky_day',
  'clearsky_night',
  'clearsky_polartwilight',
  'fair_day',
  'fair_night',
  'fair_polartwilight',
  'lightssnowshowersandthunder_day',
  'lightssnowshowersandthunder_night',
  'lightssnowshowersandthunder_polartwilight',
  'lightsnowshowers_day',
  'lightsnowshowers_night',
  'lightsnowshowers_polartwilight',
  'heavyrainandthunder',
  'heavysnowandthunder',
  'rainandthunder',
  'heavysleetshowersandthunder_day',
  'heavysleetshowersandthunder_night',
  'heavysleetshowersandthunder_polartwilight',
  'heavysnow',
  'heavyrainshowers_day',
  'heavyrainshowers_night',
  'heavyrainshowers_polartwilight',
  'lightsleet',
  'heavyrain',
  'lightrainshowers_day',
  'lightrainshowers_night',
  'lightrainshowers_polartwilight',
  'heavysleetshowers_day',
  'heavysleetshowers_night',
  'heavysleetshowers_polartwilight',
  'lightsleetshowers_day',
  'lightsleetshowers_night',
  'lightsleetshowers_polartwilight',
  'snow',
  'heavyrainshowersandthunder_day',
  'heavyrainshowersandthunder_night',
  'heavyrainshowersandthunder_polartwilight',
  'snowshowers_day',
  'snowshowers_night',
  'snowshowers_polartwilight',
  'fog',
  'snowshowersandthunder_day',
  'snowshowersandthunder_night',
  'snowshowersandthunder_polartwilight',
  'lightsnowandthunder',
  'heavysleetandthunder',
  'lightrain',
  'rainshowersandthunder_day',
  'rainshowersandthunder_night',
  'rainshowersandthunder_polartwilight',
  'rain',
  'lightsnow',
  'lightrainshowersandthunder_day',
  'lightrainshowersandthunder_night',
  'lightrainshowersandthunder_polartwilight',
  'heavysleet',
  'sleetandthunder',
  'lightrainandthunder',
  'sleet',
  'lightssleetshowersandthunder_day',
  'lightssleetshowersandthunder_night',
  'lightssleetshowersandthunder_polartwilight',
  'lightsleetandthunder',
  'partlycloudy_day',
  'partlycloudy_night',
  'partlycloudy_polartwilight',
  'sleetshowersandthunder_day',
  'sleetshowersandthunder_night',
  'sleetshowersandthunder_polartwilight',
  'rainshowers_day',
  'rainshowers_night',
  'rainshowers_polartwilight',
  'snowandthunder',
  'sleetshowers_day',
  'sleetshowers_night',
  'sleetshowers_polartwilight',
  'cloudy',
  'heavysnowshowersandthunder_day',
  'heavysnowshowersandthunder_night',
  'heavysnowshowersandthunder_polartwilight',
  'heavysnowshowers_day',
  'heavysnowshowers_night',
  'heavysnowshowers_polartwilight',
}

export type WeatherSymbols = keyof typeof WeatherSymbol;
