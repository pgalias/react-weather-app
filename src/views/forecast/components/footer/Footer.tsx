import React, { FunctionComponent } from 'react';
import { ForecastDetails, ForecastUnits } from '../../../../models';
import styles from './footer.module.css';
import { Info } from '../info/Info';
import { WindIcon } from '../wind-icon/WindIcon';

type Props = {
  units: ForecastUnits;
  forecastDetails: ForecastDetails;
};

export const Footer: FunctionComponent<Props> = ({ units, forecastDetails }) => (
  <footer className={styles.footer} role="list">
    <Info
      label="Pressure"
      value={forecastDetails.airPressureAtSeaLevel as number}
      unit={units.airPressureAtSeaLevel as string}
    />
    <Info
      label="Humidity"
      value={forecastDetails.relativeHumidity as number}
      unit={units.relativeHumidity as string}
    />
    <Info
      label="Cloudiness"
      value={forecastDetails.cloudAreaFraction as number}
      unit={units.cloudAreaFraction as string}
    />
    <Info
      label="Wind speed"
      value={forecastDetails.windSpeed as number}
      unit={units.cloudAreaFraction as string}
      icon={<WindIcon windFromDirection={forecastDetails.windFromDirection as number} />}
    />
  </footer>
);
