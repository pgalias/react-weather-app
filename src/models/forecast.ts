import { UnitsNames } from './units';
import { WeatherSymbols } from './weather-symbol';

export interface WeatherInformation {
  time: Date;
  details: Partial<Record<UnitsNames, number>>;
  symbolCode?: WeatherSymbols;
}

export interface Forecast {
  readonly units: Partial<Record<UnitsNames, string>>;
  readonly current: WeatherInformation;
}
