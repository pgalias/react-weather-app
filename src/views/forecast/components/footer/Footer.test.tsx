import React from 'react';
import { render, screen } from '@testing-library/react';
import { Footer } from './Footer';
import { ForecastDetails, ForecastUnits } from '../../../../models';

describe('Forecast::Footer', () => {
  const units: ForecastUnits = {
    windSpeed: 'km/h',
    windFromDirection: 'degrees',
    airPressureAtSeaLevel: 'hPa',
    relativeHumidity: '%',
    cloudAreaFraction: '%',
  };
  const details: ForecastDetails = {
    windSpeed: 85,
    windFromDirection: 178.2,
    airPressureAtSeaLevel: 1020,
    relativeHumidity: 70,
    cloudAreaFraction: 80,
  };

  test('should render correctly', () => {
    render(<Footer units={units} forecastDetails={details} />);

    expect(screen.getByRole('footer')).toMatchSnapshot();
  });

  test('should contain all passed information', () => {
    render(<Footer units={units} forecastDetails={details} />);
  });
});
