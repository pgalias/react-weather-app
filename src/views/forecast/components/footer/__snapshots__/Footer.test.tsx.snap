// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast::Footer should render correctly 1`] = `
<footer
  class="footer"
  role="list"
>
  <p
    class="info"
    role="listitem"
  >
    <span
      class="label"
    >
      Pressure
    </span>
    <span
      class="value"
    >
      1020
    </span>
    <span
      class="unit"
    >
      hPa
    </span>
  </p>
  <p
    class="info"
    role="listitem"
  >
    <span
      class="label"
    >
      Humidity
    </span>
    <span
      class="value"
    >
      70
    </span>
    <span
      class="unit"
    >
      %
    </span>
  </p>
  <p
    class="info"
    role="listitem"
  >
    <span
      class="label"
    >
      Cloudiness
    </span>
    <span
      class="value"
    >
      80
    </span>
    <span
      class="unit"
    >
      %
    </span>
  </p>
  <p
    class="info"
    role="listitem"
  >
    <span
      class="label"
    >
      Wind speed
    </span>
    <span
      class="value"
    >
      85
    </span>
    <span
      class="unit"
    >
      %
      <svg>
        up-left-arrow.svg
      </svg>
    </span>
  </p>
</footer>
`;
